### ISSUE TITLE
Add support for nix packages

### ISSUE THREAD

**Issue by pombredanne** [nexB/scancode-toolkit#2823]
We should add support for https://nixos.org/ packages
An example of a nix "derivation" is at https://github.com/nix-community/nix-direnv/blob/master/flake.nix
This is like a package manifest.
For instance, this is one of the packages of ScanCode in nix:
https://github.com/NixOS/nixpkgs/blob/nixos-21.11/pkgs/development/python-modules/pyahocorasick/default.nix#L29

Of note: it uses its own "language" for license declaration. We may need a simple pygmars-based nix parser. Check also https://framagit.org/upt/upt-nix/-/blob/master/upt_nix/upt_nix.py

Some other pointers include:
- https://github.com/pombredanne/nix-derivation-tool

**Comment by adityasangave** [comment]
Nixos has its own packages and we need to add support for scanning those packages am I right? I would like to work to work on these as I am familiar with package scanning in scancode can you please tell me where should I be starting

**Comment by pombredanne** [comment]
@adii21-Ux yes!... so the first step would be to document which files and index exist in a nix package that could be used to scan and collect details.
You should consider:
- A. where to collect a list of packages (e.g. an index) and what is the format
- B. what are the files/formats used in development, e.g. as seen in a VCS repo checkout 
- C. what are the files/formats used in built packages e.g. possibly included in a built binary tarball 
- D. what are the files/formats used in installed packages e.g. as seen in a installed package layout, either at the single package level or for an installed package DB

There may be subtle differences between nix and nixos there.

Once this is done, please document this either here or in a PR as comment in the to be nix.py so we can discuss it

The next steps will be to assemble a few different examples of the key file formats, starting with the most likely to be found in the wild, likely B. as .nix files and use these as test files (to be committed) 

Then equipped with these test files, you can start crafting a few basic classes. Likely starting with function to determine what is a file is  nix manifest.  I suggest you hook with @AyanSinhaMahapatra for this.
And you will likely need to recognize and parser .nix. @JonoYang is the master to ask for help.

Or course, you want to craft tests as you go.

Note that the models for data packages are going through major surgeries as we speak so it will be useful to create small PR and rebasing is likely to be required frequently.
